Intercept (dock with moving targets)

EndrosG opened the pull request:
relates to: #2356 
In my local build it works like a charm.
Fighters can dock (my flagship) at "any speed" because they are now told to adjust to the carrier's velocity vector.
This PR does not modify the general behaviour of the carrier ships, i.e. AI controlled carriers (own escorts and NPCs) still come to a full stop. This should be done in a separate PR, imho. This is just a required step...

endless-sky commented on source/AI.cpp:
```diff
@@ -1773,16 +1775,17 @@ void AI::DoScatter(Ship &ship, Command &command)
 {
 	const Point &position = ship.Position();
-	const Point &velocity = ship.Velocity();
+	const Point &velocity = ship.Velocity() - targetVelocity;
```
Taking a reference to a temporary value is an error; I'm not sure why your compiler wouldn't flag this.

EndrosG replied:
It doesnt flag it, it compiles without any warnings or alike.
As I wrote a few days ago, I am new to C++, lacking experience in objects and classes, and thus can make mistakes like these. And I am not even a native english speaker, to add on top. :-(
Would you please (simply) fix it for me?

endless-sky replied:
I would just say `Point velocity =` instead of `const Point &velocity =`.

endless-sky commented on source/AI.cpp:
```diff
@@ -1813,7 +1816,6 @@ Point AI::StoppingPoint(const Ship &ship, bool &shouldReverse)
 }
 
 
-
```
Please keep the existing whitespace between functions (3 blank lines).

endless-sky commented on source/AI.cpp:
```diff
@@ -1773,16 +1775,17 @@ void AI::DoScatter(Ship &ship, Command &command)
 
-Point AI::StoppingPoint(const Ship &ship, bool &shouldReverse)
+// Instead of coming to a full stop, adjust to a target velocity vector
+Point AI::AdjustPoint(const Ship &ship, const Point &targetVelocity, bool &shouldReverse)
```
I don't think "AdjustPoint()" is a very clear name for this function. Maybe just keep the old name of "StoppingPoint()"? Or maybe you can think of a better name.

EndrosG replied:
I can switch back to StoppingPoint() if you want. Originally I was working with 2 functions, as you can see from my commits, and then did a code cleanup afterwards by moving all calls of StoppingPoint to the new version with the targetVelocity included.
If you want me to change it back to AdjustPoint, I will happily do so tomorrow.
And then I will fix that missing 3rd line. I really thought I had 3 lines, since I was aware of that coding standard.

endless-sky replied:
I do think just keeping the StoppingPoint() name would be clearest, even though it technically isn't coming to an absolute stop anymore.

EndrosG commented:
I implemented all 3 changes.
- changed the name back to StoppingPoint (.cpp and .h)
- changed it as you provided to Point velocity
- and added the missing empty line
